### PR TITLE
[Component] - Replace `merge` with `above`/`below`

### LINKED ISSUE #1037: Ability to set ordering of Components in Group
Essentially, allowing Z-ordering of `Component`s within a group. This is particularly useful if you have multiple plots merged together and wish to highlight a particular plot by bringing it to the front.

### PR DESCRIPTION
**Pull request by bluong**
Replacing the `merge` API with an `above`/`below` API so users can make sense of where they are placing their components in regards to layering them on top of each other.

Fixes #1037

### CODE REVIEW

**Comment by jtlan** [comment]
Whoa ... mind blown. Get @crmorford's opinion on this?

File: src/components/abstractComponent.ts [hunk 1]
```diff
@@ -410,35 +410,57 @@ export module Component {
...
+     *
+     * @param {Component} c The component to merge in.
+     * @returns {ComponentGroup} The relevant ComponentGroup out of the above four cases.
+     */
+    public above(c: AbstractComponent): Component.Group {
+      return this._merge(c, false);
+    }
+
+    /**
+     * Merges this Component above another Component, returning a
```
Comment (jtlan) [inline src/components/abstractComponent.ts]: Wait, when we have `a.below(b)`, does `a` wind up below `b`, or above it? This description suggests that `a` winds up above `b`, which seems counterintuitive.

File: src/components/abstractComponent.ts [hunk 2]
```diff
@@ -410,35 +410,57 @@ export module Component {
...
       } else {
-        cg = new Plottable.Component.Group([this, c]);
+        var mergedComponents = below ? [this, c] : [c, this];
+        cg = new Plottable.Component.Group(mergedComponents);
         return cg;
       }
     }
 
     /**
+     * Merges this Component below another Component, returning a
```
Comment (jtlan) [inline src/components/abstractComponent.ts]: Wait, when we have `a.above(b)`, does `a` wind up above `b`, or below it? This description suggests that `a` winds up below `b`, which seems counterintuitive.

File: src/components/abstractComponent.ts (hunk 1 above)
Comment (bluong) [inline src/components/abstractComponent.ts]: `a` should end up below `b`  I think the description I placed is incorrect

Comment (bluong) [inline src/components/abstractComponent.ts]: Yep I accidentally swapped the two docs somehow -.-;

**Comment by jtlan** [comment]
Can you fix the description above the constructor in `Group` as well? It'd also be good to add a note about what the ordering is if you pass an array to the constructor.

File: src/components/componentGroup.ts [hunk 3]
```diff
@@ -8,8 +8,11 @@ export module Component {
      * Constructs a GroupComponent.
      *
      * A GroupComponent is a set of Components that will be rendered on top of
-     * each other. When you call Component.merge(Component), it creates and
-     * returns a GroupComponent.
+     * each other. When you call Component.above(Component) or Component.below(Component),
+     * it creates and returns a GroupComponent.
```
Comment (jtlan) [inline src/components/componentGroup.ts]: `Group` instead of `GroupComponent`, here and twice above.

**Pull request merged**